File: Haskell/media.hs
module Main(main)where
import System.IO(stdout,hSetBuffering,BufferMode(NoBuffering))

main::IO()
main = do putStrLn "Cálculo da média aritemética"
          putStrLn "-------------------------------"
          putStrLn "Digite uma sequência de números (um por linha)"
          putStrLn "Para terminar digite um valor negativo"
          lista<-leNumeros

          let
              tamanho = size lista
              media = (soma lista)/tamanho
              
          if tamanho>0
              then do putStr "média = "
                      print media
              else putStr "não existe média"
       where
           size [] = 0
           size (x:xs)= 1 + (size xs)

leNumeros = do x<-readLn::IO Double
               if x<0
                   then return []
                   else do xs<-leNumeros
                           return (x:xs)

soma [] = 0
soma (x:xs) = x + soma(xs)

File: C/media.c
#include<stdio.h>

int main(){
	float soma=0;
	int n;
	float x;
	float lista[10000];
	
	printf("Calculo da media aritemetica\n");
    printf("-------------------------------\n");
    printf("Digite uma sequencia de numeros (um por linha)\n");
    printf("Para terminar digite um valor negativo\n");

    for(n=0;;n++){
    	scanf("%f",&x);
    	if(x<0){
    		break;
    	}
    	lista[n]=x;
    }

	int i;
    for(i=0;i<n;i++){
    	soma+=lista[i];
    }
    
    if(n>0){
    	printf("media = %f\n",soma/n);   
    }
    else{
        printf("nao existe media\n");
    }

    return 0;
}
